src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function checkOnError(errorMessage: string, ...args: number[]): void {
+  if (args.some((arg: number) => arg <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkOnError('Triangle side length must be greater than zero', a, b, c);
+
+    if (a + b <= c || c + b <= a || c + a <= b) {
+      throw new Error(`
+        throws an error: sides ${a}, ${b} and ${c} can’t form a triangle
+      `);
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
```

Comment:
Same as with the Triangle constructor, you don't need to manually assign the constructor parameter to the class property when using the 'public' keyword.